### PR TITLE
chore: add k8s-do Terraform module

### PR DESCRIPTION
**Pull request by smoya**
**Description**

This PR adds a Terraform module that creates a managed K8s cluster in Digital Ocean so we can host our services there.

This Terraform module was created as part of the event-gateway-demo work, so I'm moving it from https://github.com/asyncapi/event-gateway/tree/master/deployments/asyncapi-k8s-demo-cluster to this repo.

### CODE REVIEW

File: k8s-do/main.tf [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+resource "digitalocean_kubernetes_cluster" "asyncapi-k8s-cluster" {
+  name   = var.cluster_name
+  region = var.region
+  version = var.cluster_version
+
+  node_pool {
+    name       = var.cluster_node_pool_name
+    size       = var.cluster_node_size
+    auto_scale = true
```
Comment (magicmatatjahu) [inline k8s-do/main.tf:23]: Auto scaling should be made as default? Maybe made it as customizable.

**Review by magicmatatjahu — COMMENTED** [review]
I never worked with terraform but looking at the files and their architecture they look good :) I have only one comment.

File: k8s-do/main.tf (hunk 1 above)
Comment (smoya) [reply k8s-do/main.tf]: Totally good catch! Fixed!

File: k8s-do/variables.tf [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Set the variable value in *.tfvars file
+# or using -var="do_token=..." CLI option
+variable "do_token" {}
```
Comment (magicmatatjahu) [inline k8s-do/variables.tf:3]: I think that `do_token` should be of `string` type :)

Comment (smoya) [reply k8s-do/variables.tf:3]: It is actually. Defaults to string always. Also followed the example at their official docs: https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs

Comment (magicmatatjahu) [reply k8s-do/variables.tf:3]: Thanks!

**Review by magicmatatjahu — APPROVED** [review]

**Pull request merged**